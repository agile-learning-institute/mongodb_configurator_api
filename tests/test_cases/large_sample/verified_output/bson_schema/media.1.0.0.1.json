{
  "description": "A media item in the system",
  "bsonType": "object",
  "properties": {
    "_id": {
      "description": "The unique identifier for the media",
      "bsonType": "objectId"
    },
    "title": {
      "description": "The title of the media",
      "pattern": "^[^\\t\\n\\r]{0,255}$",
      "bsonType": "string"
    },
    "type": {
      "description": "The type of media",
      "bsonType": "string",
      "enum": [
        "movie",
        "tv_show",
        "documentary",
        "short"
      ]
    },
    "status": {
      "description": "The current status of the media",
      "bsonType": "string",
      "enum": [
        "draft",
        "published",
        "archived"
      ]
    },
    "last_saved": {
      "description": "When the media was last updated",
      "bsonType": "object",
      "properties": {
        "from_ip": {
          "description": "Http Request remote IP address",
          "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
          "bsonType": "string"
        },
        "by_user": {
          "description": "ID Of User",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        },
        "at_time": {
          "description": "The date-time when last updated",
          "bsonType": "date"
        },
        "correlation_id": {
          "description": "The logging correlation ID of the update transaction",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        }
      },
      "additionalProperties": false,
      "required": [
        "from_ip",
        "by_user",
        "at_time",
        "correlation_id"
      ]
    },
    "tags": {
      "description": "Tags associated with the media",
      "bsonType": "array",
      "items": {
        "bsonType": "string",
        "enum": [
          "action",
          "comedy",
          "drama",
          "sci_fi",
          "documentary"
        ]
      }
    },
    "metadata": {
      "description": "Additional metadata about the media",
      "bsonType": "object",
      "properties": {
        "duration": {
          "description": "Duration in minutes",
          "bsonType": "int",
          "minimum": 1
        },
        "format": {
          "description": "Media format",
          "bsonType": "string",
          "enum": [
            "dvd",
            "bluray",
            "digital",
            "streaming"
          ]
        },
        "quality": {
          "description": "Quality rating",
          "bsonType": "string",
          "enum": [
            "sd",
            "hd",
            "uhd"
          ]
        }
      },
      "additionalProperties": false
    }
  },
  "required": [
    "_id",
    "title",
    "status",
    "last_saved"
  ],
  "additionalProperties": false
}