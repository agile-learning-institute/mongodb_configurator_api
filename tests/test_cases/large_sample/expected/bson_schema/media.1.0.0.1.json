{
  "bsonType": "object",
  "required": ["_id", "title", "status", "last_saved"],
  "properties": {
    "_id": {
      "bsonType": "objectId",
      "pattern": "^[0-9a-fA-F]{24}$"
    },
    "title": {
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
    },
    "type": {
      "bsonType": "string",
      "enum": ["image", "video", "audio", "document"]
    },
    "status": {
      "bsonType": "string",
      "enum": ["draft", "active", "archived"]
    },
    "last_saved": {
      "bsonType": "object",
      "required": ["from_ip", "by_user", "at_time", "correlation_id"],
      "properties": {
        "from_ip": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "by_user": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "at_time": {
          "bsonType": "date"
        },
        "correlation_id": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        }
      }
    },
    "tags": {
      "bsonType": "array",
      "items": {
        "bsonType": "string",
        "enum": ["featured", "popular", "trending", "new", "updated"]
      }
    },
    "metadata": {
      "bsonType": "object",
      "properties": {
        "duration": {
          "bsonType": "int",
          "minimum": 0
        },
        "format": {
          "bsonType": "string",
          "enum": ["mp4", "mp3", "jpg", "png", "pdf"]
        },
        "quality": {
          "bsonType": "string",
          "enum": ["low", "medium", "high", "ultra"]
        }
      }
    }
  }
}
